packages/go/cypher/model/pgsql/translate/translation_cases/node.sql

```diff
@@ -0,0 +1,45 @@
+-- case: match (s) return s
+with n1 as (select n1.* from node n1)
+select (n1.id, n1.kind_ids, n1.properties)::nodecomposite as s
+from n1;
+
+-- case: match (s) where s.name = '1234' return s
+with n1 as (select n1.* from node n1 where n1.properties -> 'name' = '1234')
+select (n1.id, n1.kind_ids, n1.properties)::nodecomposite as s
+from n1;
+
+-- case: match (s), (e) where s.name = '1234' return s
+with n1 as (select n1.* from node n1 where n1.properties -> 'name' = '1234'),
+     n2 as (select n2.* from node n2)
+select (n1.id, n1.kind_ids, n1.properties)::nodecomposite as s
+from n1;
+
+-- case: match (s:A), (e:B) where s.name = e.name return s, e
+with n1 as (select n1.* from node n1 where n1.kind_ids operator (pg_catalog.&&) array []::int2[]),
```

Comment:
i haven't looked into the translator implementation for node labels yet, but wanted to make sure that you expect the binary expression in this where clause:

`n1.kind_ids operator (pg_catalog.&&) array []::int2[]`

to be an empty array?  logically this seems like its probably fine, because the node label "A" isn't an actual BH kind and im guessing whatever translates this clause under the hood is returning an empty array accordingly.

Reply:
Kind mappings require a live database so I haven't gotten around to plugging that particular interface into the formatter. Leaving this unresolved to review once the KindMapper type is hoisted into the formatter function logic.